src/geometry/TorusSegmentGroup.ts

```diff
@@ -95,29 +42,33 @@ export default class TorusSegmentGroup extends PlaneGroup {
     arcAngle: number,
     filterOptions?: FilterOptions,
   ) {
-    this.setNodeId(nodeId, this.count);
-    this.setTreeIndex(treeIndex, this.count);
-    this.setColor(color, this.count);
-    this.setCenter(center, this.count);
-    this.setNormal(normal, this.count);
-    this.setRadius(radius, this.count);
-    this.setTubeRadius(tubeRadius, this.count);
-    this.setAngle(angle, this.count);
-    this.setArcAngle(arcAngle, this.count);
-    this.count += 1;
+    this.setNodeId(nodeId, this.data.count);
+    this.setTreeIndex(treeIndex, this.data.count);
+    this.setColor(color, this.data.count);
+    this.data.add({
+      nodeId: nodeId,
```

Comment:
When the variable you set as value is named the same as the key, you can simply do
```
this.data.add({
      nodeId,
      ...
});

Reply:
Fixed